Bas plan init on terminate

bjoernschroeder opened the pull request.

bjoernschroeder committed:
add createPlanCreator and createDomainPlan to interface

bjoernschroeder committed:
implement createPlanCreator and createDomainPlan

bjoernschroeder committed:
add templates for DomainPlan and PlanCreator, add class to plans

bjoernschroeder committed:
generate DomainPlan and PlanCreator

bjoernschroeder committed:
add id to plan includes, fix typos

bjoernschroeder committed:
add constructor/destructor implementation for plans

bjoernschroeder committed:
use planname + id for classname

bjoernschroeder committed:
add protected region for method implementation, fix typo

bjoernschroeder committed:
remove prints

StephanOpfer reviewed (approved):
@abhi1287 I think we will need to address the code generation sooner or later in general. 

- I think with the idea from Alex we can get rid of the protected regions
- Not all methods/classes need to be generated every time. E.g. *constraint.h/.cpp are unecessary if there are no variables attached to the corresponding conditions.

bjoernschroeder committed:
remove name from DomainPlans, use unique_ptrs in creators

bjoernschroeder committed:
generate run method for plans

bjoernschroeder committed:
fix NullPointer

bjoernschroeder committed:
generate run method in header

bjoernschroeder committed:
fix run method generation

veerajsk committed:
Merge pull request #10 from rapyuta-robotics/bas_plan_run

generate run method for plans

bjoernschroeder committed:
remove run stub from plans

bjoernschroeder committed:
add onInit and onTerminate to plan header

bjoernschroeder committed:
update plan header generation

veerajsk committed:
Merge pull request #11 from rapyuta-robotics/bas_remove_run_stubs

Remove_run_stub_add_onInit_onTerminate_stub

veerajsk committed:
Fix protected region code generation for basic plan

Prevent indefinitely adding the "override these methods for your use case ..." in the generated code

veerajsk committed:
Improve comment
